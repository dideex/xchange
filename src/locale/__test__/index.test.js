import React from 'react'
import Component from '../index'
import {shallow} from 'enzyme'
import {FormattedMessage} from 'react-intl'

import UserStore from '../../store/User'

describe('Locale', () => {
  let userStore = new UserStore()
  beforeEach(() => {
    userStore = new UserStore()
  })
  it('Should render without crashes', () => {
    const wrapper = shallow(<Component userStore={userStore} />)
    wrapper.unmount()
  })
  describe('locales', () => {
    it('Russian locale should exist', () => {
      userStore.locale = 'ru-RU'
      const wrapper = shallow(
        <Component userStore={userStore}>
          <FormattedMessage id="test" />
        </Component>,
      )
      
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('English locale should exist', () => {
      userStore.locale = 'en-US'
      const wrapper = shallow(
        <Component userStore={userStore}>
          <FormattedMessage id="test" />
        </Component>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('France locale should exist', () => {
      userStore.locale = 'fr-FR'
      const wrapper = shallow(
        <Component userStore={userStore}>
          <FormattedMessage id="test" />
        </Component>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Chainise locale should exist', () => {
      userStore.locale = 'zh-ZH'
      const wrapper = shallow(
        <Component userStore={userStore}>
          <FormattedMessage id="test" />
        </Component>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
