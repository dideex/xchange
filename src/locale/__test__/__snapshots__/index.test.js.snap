// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locale locales Chainise locale should exist 1`] = `"<span>测试</span>"`;

exports[`Locale locales English locale should exist 1`] = `"<span>test</span>"`;

exports[`Locale locales France locale should exist 1`] = `"<span>tester</span>"`;

exports[`Locale locales Russian locale should exist 1`] = `"<span>тест</span>"`;
